ivy/utils/backend/sub_backend_handler.py

```diff
@@ -200,3 +210,65 @@ def find_available_sub_backends(sub_backends_loc):
             available_sub_backends.append(sub_backend)
 
     return available_sub_backends
+
+
+def _find_available_sub_backend_implementations(sub_backends):
+    result = dict()
+    for sub in sub_backends:
+        sub_backend = ivy.utils.dynamic_import.import_module(_sub_backend_dict[sub])
+        for k, v in sub_backend.__dict__.items():
+            if isinstance(v, Callable) and not k.startswith("__"):
+                result[k] = result.get(k, []) + [sub]
+
+    return result
+
+
+def available_sub_backend_implementations(obj: Union[Callable, str]) -> list:
+    """
+    Return whether a sub-backend implementation is available for `obj`.
+
+    Parameters
+    ----------
+    obj : callable or str
+        the object for which to check if a sub-backend implementation is available.
+
+    Returns
+    -------
+    ret : list
+        a list of sub-backend implementations available for `obj`.
+
+    Examples
+    --------
+    >>> import ivy
+    >>> ivy.set_backend('torch')
+    >>> ivy.available_sub_backend_implementations(ivy.scaled_dot_product_attention)
+    ['xformers']
+    >>> ivy.set_backend('numpy')
+    >>> ivy.available_sub_backend_implementations(ivy.scaled_dot_product_attention)
+    []
+    """
+    obj = _check_callable(obj)
+    sub_backends = ivy.current_backend().available_sub_backends()
+    result = []
+    if not sub_backends:
+        return result
+    if (
+        ivy.current_backend_str()
+        not in _available_sub_backends_implementations_dict.keys()
+    ):
+        _available_sub_backends_implementations_dict[ivy.current_backend_str()] = (
+            _find_available_sub_backend_implementations(sub_backends)
+        )
+    return _available_sub_backends_implementations_dict[ivy.current_backend_str()].get(
+        obj.__name__, []
+    )
+
+
+def _check_callable(obj):
```

Comment:
we wouldn't need this helper after changing `obj` to only be a `str` in the `available_sub_backend_implementations` function